src/devices/Amg88xx/Amg88xx.cs

```diff
@@ -0,0 +1,475 @@
+        /// </summary>
+        /// <param name="state">True, to switch moving average on</param>
+        public void SetMovingAverageModeState(bool state)
+        {
+            SetBit(Register.AVE, (byte)MovingAverageModeBit.MAMOD, state);
+        }
+
+        #endregion
+
+        #region Frame Rate
+
+        /// <summary>
+        /// Get the current frame rate.
+        /// </summary>
+        /// <returns>Frame rate (either 1 or 10fps) </returns>
+        public int GetFrameRate()
+        {
+            return GetBit(Register.FPSC, (byte)FrameRateBit.FPS) ? 1 : 10;
+        }
+
+        /// <summary>
+        /// Sets the frame rate (either 1 or 10fps).
+        /// </summary>
+        /// <param name="frameRate">Frame rate</param>
+        /// <exception cref="ArgumentException">Thrown when attempting to set a frame rate other than 1 or 10</exception>
+        public void SetFrameRate(int frameRate)
+        {
+            if (frameRate != 1 && frameRate != 10)
+            {
+                throw new ArgumentException("Frame rate must either be 1 or 10.", nameof(frameRate));
+            }
+
+            SetBit(Register.FPSC, (byte)FrameRateBit.FPS, frameRate == 1);
+        }
+        #endregion
+
+        #region Operating Mode / Power Control
+
+        /// <summary>
+        /// Gets the current operating mode
+        /// </summary>
+        /// <returns>Operating mode</returns>
+        public OperatingMode GetOperatingMode()
+        {
+            return (OperatingMode)GetRegister(Register.PCLT);
+        }
+
+        /// <summary>
+        /// Sets the operating mode
+        /// </summary>
+        /// <param name="operatingMode">Operating mode</param>
+        public void SetOperatingMode(OperatingMode operatingMode)
+        {
+            SetRegister(Register.PCLT, (byte)operatingMode);
+        }
+        #endregion
+
+        #region Reset
+
+        /// <summary>
+        /// Performs an initial reset of the sensor. The flags and all configuration registers
+        /// are reset to default values.
+        /// </summary>
+        public void InitialReset()
+        {
+            // an initial reset (factory defaults) is initiated by writing 0x3f into the reset register (RST)
+            SetRegister(Register.RST, (byte)ResetType.Initial);
+        }
+
+        /// <summary>
+        /// Performs a reset of all flags
+        /// </summary>
+        public void FlagReset()
+        {
+            // a reset of all flags (status register, interrupt flag and interrupt table) is initiated by writing 0x30
+            // into the reset register (RST)
+            SetRegister(Register.RST, (byte)ResetType.Flag);
+        }
+
+        #endregion
+
+        #region Interrupt control, levels and pixel flags
+
+        /// <summary>
+        /// Gets the interrupt mode
+        /// </summary>
+        /// <returns>Interrupt mode</returns>
+        public InterruptMode GetInterruptMode()
```

Comment:
Would go for a property as well `InterruptMode`

Reply:
Done